### PR TITLE
Adapted auto complete page view to declarative shadow DOM. #97

### PR DESCRIPTION
**Pull request by PatrykSzwer**
PR for declarative shadow DOM adaptation changes. Not mergable yet.

### CODE REVIEW

File: src/KitchenSink/wwwroot/KitchenSink/AutocompletePage.html [hunk 1]
```diff
@@ -1,34 +1,26 @@
...
-            </p>
-            <input type="text" class="form-control" value="{{model.PlacesSearch$::input}}" placeholder="Poland? Sweden?" on-blur="clearPlaces" />
-            <ul class="kitchensink-autocomplete">
-                <template is="dom-repeat" items="{{model.FoundPlaces}}">
-                    <li on-mousedown="selectPlace" class="kitchensink-autocomplete__item" item="{{item}}">{{item.Name}}</li>
-                </template>
-            </ul>
-            <p id="kitchensink-autocomplete-capital">{{model.PlacesText}}</p>
-        </div>
+        <input slot="kitchensink/autocomplete-products-input" id="kitchensink-autocomplete-products-input" type="text" class="form-control" value="{{model.ProductsSearch$::input}}" placeholder="Whiskey? Whisky?" on-blur="clearProducts" />
```
Comment (Mackiovello) [inline src/KitchenSink/wwwroot/KitchenSink/AutocompletePage.html]: It's, in general, not a good idea to use id with Starcounter apps. I'm not sure if it applies to this case. @tomalec

File: src/KitchenSink/wwwroot/KitchenSink/AutocompletePage.html [hunk 2]
```diff
@@ -58,4 +50,20 @@
...
     </script>
+    <template is="declarative-shadow-dom">
+        <slot name="kitchensink/autocomplete-heading"></slot>
+        <div title="products">
+            <slot name="kitchensink/autocomplete-products-description"></slot>
+            <slot name="kitchensink/autocomplete-products-input"></slot>
+            <slot name="kitchensink/autocomplete-products-autocomplete"></slot>
+            <slot name="kitchensink/autocomplete-products-price"></slot>
+        </div>
+        <br />
```
Comment (Mackiovello) [inline src/KitchenSink/wwwroot/KitchenSink/AutocompletePage.html]: Even if you didn't put it there, `<br/>` shouldn't be used like this. Could you exchange it for some margin instead?

File: src/KitchenSink/wwwroot/KitchenSink/AutocompletePage.html (hunk 1 above)
Comment (tomalec) [inline src/KitchenSink/wwwroot/KitchenSink/AutocompletePage.html]: Definitely it shouldn't event in this case, you could imagine somebody mapping this page to two different places in the document. For example in "Two players app" which makes every page look like good old split-screen game, so two people could fiddle with it simultaneously on the same screen. 

Then your id is not unique, what violates HTML rules, and may mak some scripts crash.

File: src/KitchenSink/wwwroot/KitchenSink/AutocompletePage.html [hunk 3]
```diff
@@ -1,34 +1,26 @@
...
+        <ul slot="kitchensink/autocomplete-products-autocomplete" class="kitchensink-autocomplete">
+            <template is="dom-repeat" items="{{model.FoundProducts}}">
+                <!-- Blur fires before click, but after mousedown -->
+                <li on-mousedown="selectProduct" class="kitchensink-autocomplete__item kitchensink-test-autocomplete-products-item" item="{{item}}">{{item.Name}}</li>
+            </template>
+        </ul>
+        <p slot="kitchensink/autocomplete-products-price" id="kitchensink-autocomplete-price">{{model.ProductsText}}</p>
+
+        <p slot="kitchensink/autocomplete-places-description" class="lead">This autocomplete field uses pre-defined list of places (although it's usually better to use DB as source).</p>
+        <input slot="kitchensink/autocomplete-places-input" id="kitchensink-autocomplete-places-input" type="text" class="form-control" value="{{model.PlacesSearch$::input}}" placeholder="Poland? Sweden?" on-blur="clearPlaces" />
```
Comment (tomalec) [inline src/KitchenSink/wwwroot/KitchenSink/AutocompletePage.html]: Same story with `id` as above.

File: src/KitchenSink/wwwroot/KitchenSink/AutocompletePage.html [hunk 4]
```diff
@@ -1,34 +1,26 @@
...
+        <ul slot="kitchensink/autocomplete-places-autocomplete" class="kitchensink-autocomplete">
+            <template is="dom-repeat" items="{{model.FoundPlaces}}">
+                <li on-mousedown="selectPlace" class="kitchensink-autocomplete__item kitchensink-test-autocomplete-places-item" item="{{item}}">{{item.Name}}</li>
+            </template>
+        </ul>
+        <p slot="kitchensink/autocomplete-places-capital" id="kitchensink-autocomplete-capital">{{model.PlacesText}}</p>
```
Comment (tomalec) [inline src/KitchenSink/wwwroot/KitchenSink/AutocompletePage.html]: `id` as above

File: src/KitchenSink/wwwroot/KitchenSink/AutocompletePage.html (hunk 1 above)
Comment (PatrykSzwer) [inline src/KitchenSink/wwwroot/KitchenSink/AutocompletePage.html]: Ok, good point. It was for tests selectors, I'll replace them with custom classes.

File: src/KitchenSink/wwwroot/KitchenSink/AutocompletePage.html (hunk 2 above)
Comment (PatrykSzwer) [inline src/KitchenSink/wwwroot/KitchenSink/AutocompletePage.html]: I've used margin-bottom on products element. Please take a look if this solution is okey.

Comment (Mackiovello) [inline src/KitchenSink/wwwroot/KitchenSink/AutocompletePage.html]: That's a good solution

**Comment by PatrykSzwer** [comment]
Ok, I think It's everything. PR is ready for review.

File: src/KitchenSink/wwwroot/KitchenSink/AppIcon.html [hunk 5]
```diff
@@ -1,3 +1,3 @@
 ﻿<template>
-    <img src="/KitchenSink/images/KitchenSink.svg">
+    <img slot="kitchensink/kitchensink-image" src="/KitchenSink/images/KitchenSink.svg">
```
Comment (Mackiovello) [inline src/KitchenSink/wwwroot/KitchenSink/AppIcon.html]: Please use a more descriptive slot name like "app-icon" or "icon". "Image" is a little too general

File: src/KitchenSink/wwwroot/KitchenSink/BreadcrumbPage.html [hunk 6]
```diff
@@ -3,25 +3,24 @@
...
-                <span class="btn" select="{{item.SelectTrigger$::mouseup}}" onmousedown="++this.select">{{item.Name}}</span>
+        <p slot="kitchensink/breadcrumb-simple-breadcrumb-description" class="lead">A JSON <code>array</code> of <code>objects</code> can used as a breadcrumb navigation. The below example uses <code>template is="dom-repeat"</code> with <code>span</code>.</p>
+        <div slot="kitchensink/breadcrumb-simple-breadcrumb">
+        <template is="dom-repeat" items="{{model.Breadcrumbs}}" restamp>
+            <template is="dom-if" if="{{index}}" restamp>
+                &gt;
             </template>
+            <span class="btn" select="{{item.SelectTrigger$::mouseup}}" onmousedown="++this.select">{{item.Name}}</span>
+        </template>
         </div>
```
Comment (Mackiovello) [inline src/KitchenSink/wwwroot/KitchenSink/BreadcrumbPage.html]: The indentation is wrong on this one

File: src/KitchenSink/wwwroot/KitchenSink/CallbackPage.html [hunk 7]
```diff
@@ -1,58 +1,53 @@
...
         </p>
 
-        <h4>Responsive save button</h4>
-        <p>
-            <button type="button" class="btn btn-sm btn-default" value="{{model.SaveTrigger$::click}}" disabled="{{model.SaveTrigger$}}" onmouseup="++this.value;">Save changes (1s)</button>
-            <button type="button" class="btn btn-sm btn-default" value="{{model.SaveAndSpinTrigger$::click}}" disabled="{{model.SaveAndSpinTrigger$}}" onmouseup="++this.value;">
+        <h4 slot="kitchensink/callback-responsive-label">Responsive save button</h4>
+        <button slot="kitchensink/callback-responsive-button" type="button" class="btn btn-sm btn-default" value="{{model.SaveTrigger$::click}}" disabled="{{model.SaveTrigger$}}" onmouseup="++this.value;">Save changes (1s)</button>
+            <button slot="kitchensink/callback-responsive-button-spin" type="button" class="btn btn-sm btn-default" value="{{model.SaveAndSpinTrigger$::click}}" disabled="{{model.SaveAndSpinTrigger$}}" onmouseup="++this.value;">
                 <template is="dom-if" if="{{model.SaveAndSpinTrigger$}}">
```
Comment (Mackiovello) [inline src/KitchenSink/wwwroot/KitchenSink/CallbackPage.html]: The indenting is off here

File: src/KitchenSink/wwwroot/KitchenSink/CallbackPage.html [hunk 8]
```diff
@@ -196,4 +191,26 @@
             };
         })();
     </script>
+    <template is="declarative-shadow-dom">
+        <slot name="kitchensink/callback-heading"></slot>
+        <slot name="kitchensink/callback-description"></slot>
+        <slot name="kitchensink/callback-responsive-label"></slot>
+        <p>
```
Comment (Mackiovello) [inline src/KitchenSink/wwwroot/KitchenSink/CallbackPage.html]: Why is this a `p` element and not a `div`?

File: src/KitchenSink/wwwroot/KitchenSink/DialogPage.html [hunk 9]
```diff
@@ -22,4 +22,12 @@
             </div>
         </paper-dialog>
     </template>
+    <template is="declarative-shadow-dom">
+        <slot name="kitchensink/dialog-description"></slot>
+        <p>
```
Comment (Mackiovello) [inline src/KitchenSink/wwwroot/KitchenSink/DialogPage.html]: What's the purpose of this one? Is it to make it a block level element?

File: src/KitchenSink/wwwroot/KitchenSink/FlashMessagePage.html [hunk 10]
```diff
@@ -39,4 +37,19 @@
             };
         })();
     </script>
+    <template is="declarative-shadow-dom">
+        <slot name="kitchensink/flashMessage-header"></slot>
+        <slot name="kitchensink/flashMessage-description"></slot>
+        <slot name="kitchensink/flashMessage-dom-if-description"></slot>
+        <slot name="kitchensink/flashMessage-schedule-task-description"></slot>
+        <p>
```
Comment (Mackiovello) [inline src/KitchenSink/wwwroot/KitchenSink/FlashMessagePage.html]: Again, what's the purpose of this element?

File: src/KitchenSink/wwwroot/KitchenSink/MultiselectPage.html [hunk 11]
```diff
@@ -2,30 +2,32 @@
...
                 <select class="form-control" multiple></select>
             </juicy-select>
         </p>
-        <p>
+        <p slot="kitchensink/multiselect-favorite-country-select">
             <span>Select your favorite country:</span>
             <juicy-select class="kitchensink-juicyselect" caption-text="No favorite country" value="{{model.FavoriteCountry$}}" options="{{model.Countries}}" selected-property="Selected$" text-property="Name" value-property="Name">
                 <select class="form-control"></select>
             </juicy-select>
         </p>
```
Comment (Mackiovello) [inline src/KitchenSink/wwwroot/KitchenSink/MultiselectPage.html]: The slot should be on the span and `juicy-select` elements here. It's the same case above.

File: src/KitchenSink/wwwroot/KitchenSink/NestedPartial.html [hunk 12]
```diff
@@ -1,13 +1,15 @@
...
+        <p slot="kitchensink/nested-description" class="lead">JSON's primitive value type <code>object</code> can be used for nesting of the partials. This also works for recursive nesting.</p>
 
-        <div class="kitchensink-nested-child">
+        <div slot="kitchensink/nested-nested-child" class="kitchensink-nested-child">
             <button class="btn btn-sm btn-default" value="{{model.AddChildTrigger$::click}}" onmousedown="++this.value">Add child</button>
             <div>Hi, <span>{{model.Name}}</span></div>
             <div>
                 <starcounter-include partial="{{model.ChildPartial}}"></starcounter-include>
             </div>
         </div>
```
Comment (Mackiovello) [inline src/KitchenSink/wwwroot/KitchenSink/NestedPartial.html]: The div doesn't serve a purpose here. It should be removed and the slots be put on its children

File: src/KitchenSink/wwwroot/KitchenSink/ProgressBarPage.html [hunk 13]
```diff
@@ -1,20 +1,22 @@
...
             and is continually updated throughout the progression of the task. </p>
 
-        <p> Read more about <code>Scheduling.ScheduleTask</code> and <code>Session.ScheduleTask</code> on starcounter.io - <a href="http://starcounter.io/guides/transactions/running-background-jobs/">Running background jobs</a>. </p>
+        <p slot="kitchensink/progressBar-read-more"> Read more about <code>Scheduling.ScheduleTask</code> and <code>Session.ScheduleTask</code> on starcounter.io - <a href="http://starcounter.io/guides/transactions/running-background-jobs/">Running background jobs</a>. </p>
        
-         <div class="kitchensink-progress-bar__container">
+        <div slot="kitchensink/progressBar-progress-bar" class="kitchensink-progress-bar__container">
             <span class="kitchensink-progress-bar__percentages"> {{model.Progress}}% </span>
             <div class="kitchensink-progress-bar__fill" style="width:{{model.Progress}}%"></div>
         </div>
```
Comment (Mackiovello) [inline src/KitchenSink/wwwroot/KitchenSink/ProgressBarPage.html]: The div shouldn't be here. Place the slot on the children instead

**Comment by warpech** [comment]
@Mackiovello could you pls do the final review?

File: src/KitchenSink/wwwroot/KitchenSink/DialogPage.html (hunk 9 above)
Comment (PatrykSzwer) [inline src/KitchenSink/wwwroot/KitchenSink/DialogPage.html]: Yes, It's copied from original solution, to make button looks more spaced. Do You think that I should replace this solution with divs and classes with margins?

File: src/KitchenSink/wwwroot/KitchenSink/AutocompletePage.html [hunk 14]
```diff
@@ -1,34 +1,26 @@
...
-            <ul class="kitchensink-autocomplete">
-                <template is="dom-repeat" items="{{model.FoundPlaces}}">
-                    <li on-mousedown="selectPlace" class="kitchensink-autocomplete__item" item="{{item}}">{{item.Name}}</li>
-                </template>
-            </ul>
-            <p id="kitchensink-autocomplete-capital">{{model.PlacesText}}</p>
-        </div>
+        <h1 slot="kitchensink/autocomplete-heading">Autocomplete</h1>
+
+        <p slot="kitchensink/autocomplete-products-description" class="lead">This autocomplete field uses DB as source. Enter <code>'*'</code> to show all available items</p>
```
Comment (tomalec) [inline src/KitchenSink/wwwroot/KitchenSink/AutocompletePage.html]: Please prefix the classes with app name, not to collide with other apps in light DOM

Comment (tomalec) [inline src/KitchenSink/wwwroot/KitchenSink/AutocompletePage.html]: or is that a bootstrap thingy, which we unfortunately still use?

File: src/KitchenSink/wwwroot/KitchenSink/BreadcrumbPage.html [hunk 15]
```diff
@@ -37,18 +36,20 @@
                 </x-crumb>
             </template>
         </x-breadcrumb>
-        <template is="dom-if" if="{{model.CurrentTreeItem.IsAdd}}" restamp>
+        <template slot="kitchensink/breadcrumb-adding-heading" is="dom-if" if="{{model.CurrentTreeItem.IsAdd}}" restamp>
```
Comment (tomalec) [inline src/KitchenSink/wwwroot/KitchenSink/BreadcrumbPage.html]: We need `slot="kitchensink/breadcrumb-adding-heading" ` on `<h5>` element as that is the one that would eventually be stamped.

If you want to add a slot name for `template is="dom-if"` I would call it rather  `"kitchensink/breadcrumb-adding-heading-condition"`, @warpech @Mackiovello WDYT? do we want to add explicit slot names for templates?

File: src/KitchenSink/wwwroot/KitchenSink/BreadcrumbPage.html [hunk 16]
```diff
@@ -37,18 +36,20 @@
...
             <h5>Adding a new type in <strong>{{model.CurrentTreeItem.ParentName}}</strong></h5>
         </template>
 
-        <template is="dom-if" if="{{!model.CurrentTreeItem.IsAdd}}" restamp>
+        <template slot="kitchensink/breadcrumb-editing-heading" is="dom-if" if="{{!model.CurrentTreeItem.IsAdd}}" restamp>
             <h5>Editing <strong>{{model.CurrentTreeItem.Name$}}</strong></h5>
```
Comment (tomalec) [inline src/KitchenSink/wwwroot/KitchenSink/BreadcrumbPage.html]: `h5` needs a slot names, as above

File: src/KitchenSink/wwwroot/KitchenSink/BreadcrumbPage.html [hunk 17]
```diff
@@ -37,18 +36,20 @@
...
-        <label class="control-label">Name</label>
+        <label slot="kitchensink/breadcrumb-name-heading" class="control-label">Name</label>
 
-        <input type="text" value="{{model.CurrentTreeItem.Name$::change}}" placeholder="Name" class="form-control">
+        <input slot="kitchensink/breadcrumb-name-input" type="text" value="{{model.CurrentTreeItem.Name$::change}}" placeholder="Name" class="form-control">
 
-        <button class="btn btn-sm btn-default" value="{{model.CurrentTreeItem.SaveTrigger$::click}}" onmousedown="++this.value">Save</button>
+        <button slot="kitchensink/breadcrumb-name-save" class="btn btn-sm btn-default" value="{{model.CurrentTreeItem.SaveTrigger$::click}}" onmousedown="++this.value">Save</button>
+    </template>
+    <template is="declarative-shadow-dom">
```
Comment (tomalec) [inline src/KitchenSink/wwwroot/KitchenSink/BreadcrumbPage.html]: You don't have to add it if it's empty

File: src/KitchenSink/wwwroot/KitchenSink/ButtonPage.html [hunk 18]
```diff
@@ -69,4 +66,6 @@
             };
         })();
     </script>
+    <template is="declarative-shadow-dom">
```
Comment (tomalec) [inline src/KitchenSink/wwwroot/KitchenSink/ButtonPage.html]: You don't have to add it if it's empty

File: src/KitchenSink/wwwroot/KitchenSink/ChartPage.html [hunk 19]
```diff
@@ -14,18 +14,20 @@
...
+                        pointStrokeColor: "#fff",
+                        pointHighlightFill: "#fff",
+                        pointHighlightStroke: "rgba(220,220,220,1)",
+                        data: template.model.Temperatures
+                    }
                 ]
             };
         });
     </script>
+    <template is="declarative-shadow-dom">
```
Comment (tomalec) [inline src/KitchenSink/wwwroot/KitchenSink/ChartPage.html]: You don't have to add it if it's empty

File: src/KitchenSink/wwwroot/KitchenSink/CheckboxPage.html [hunk 20]
```diff
@@ -1,11 +1,13 @@
...
+        <label slot="kitchensink/checkbox-label">Driver's license</label>
 
-    <input class="kitchensink-test-driver-license-input" type="checkbox" checked="{{model.DrivingLicense$::change}}">
+        <input slot="kitchensink/checkbox-input" class="kitchensink-test-driver-license-input" type="checkbox" checked="{{model.DrivingLicense$::change}}">
 
-    <div class="kitchensink-test-driver-license-label">{{model.DrivingLicenseReaction}}</div>
-  </template>
+        <div slot="kitchensink/checkbox-license-label" class="kitchensink-test-driver-license-label">{{model.DrivingLicenseReaction}}</div>
+    </template>
+    <template is="declarative-shadow-dom">
```
Comment (tomalec) [inline src/KitchenSink/wwwroot/KitchenSink/CheckboxPage.html]: You don't have to add it if it's empty

File: src/KitchenSink/wwwroot/KitchenSink/CookiePage.html [hunk 21]
```diff
@@ -32,4 +32,6 @@
             };
         })();
     </script>
+    <template is="declarative-shadow-dom">
```
Comment (tomalec) [inline src/KitchenSink/wwwroot/KitchenSink/CookiePage.html]: You don't have to add it if it's empty

File: src/KitchenSink/wwwroot/KitchenSink/AutocompletePage.html (hunk 14 above)
Comment (tomalec) [inline src/KitchenSink/wwwroot/KitchenSink/AutocompletePage.html]: I'd rather try to style it in Shadow DOM instead.

File: src/KitchenSink/wwwroot/KitchenSink/DialogPage.html (hunk 9 above)
Comment (Mackiovello) [inline src/KitchenSink/wwwroot/KitchenSink/DialogPage.html]: It's no panic, using `p` is just semantically wrong. It seems the `p` element has been used throughout the project in order to add some margin. I would consider it a separate issue and focus on the declarative-shadow-dom stuff right now

File: src/KitchenSink/wwwroot/KitchenSink/AutocompletePage.html (hunk 14 above)
Comment (Mackiovello) [inline src/KitchenSink/wwwroot/KitchenSink/AutocompletePage.html]: `lead` is a bootstrap class. Are we actually planning to move from bootstrap?

File: src/KitchenSink/wwwroot/KitchenSink/BreadcrumbPage.html (hunk 15 above)
Comment (Mackiovello) [inline src/KitchenSink/wwwroot/KitchenSink/BreadcrumbPage.html]: What's been done previously, to my knowledge, is to wrap the `template` in a `div` and put the slot on that one or put the slot on the child as you suggest. In this case, it seems reasonable to put the slot on the `h5`. 

I'm pretty sure there was some kind of issue with putting slots on `template` elements, which is why that was never done in the People app

Comment (PatrykSzwer) [inline src/KitchenSink/wwwroot/KitchenSink/BreadcrumbPage.html]: Yes, @Mackiovello is right, I've always done that div wrap in next files, in this one I didn't know that yet and forgot to change it later. I'll fix it.

File: src/KitchenSink/wwwroot/KitchenSink/BreadcrumbPage.html (hunk 17 above)
Comment (PatrykSzwer) [inline src/KitchenSink/wwwroot/KitchenSink/BreadcrumbPage.html]: In #97 We talk about that and It's good practice to add it even if unnecessary, for future changes.

File: src/KitchenSink/wwwroot/KitchenSink/ButtonPage.html (hunk 18 above)
Comment (PatrykSzwer) [inline src/KitchenSink/wwwroot/KitchenSink/ButtonPage.html]: Same as above.

File: src/KitchenSink/wwwroot/KitchenSink/ChartPage.html (hunk 19 above)
Comment (PatrykSzwer) [inline src/KitchenSink/wwwroot/KitchenSink/ChartPage.html]: Same as above.

File: src/KitchenSink/wwwroot/KitchenSink/CheckboxPage.html (hunk 20 above)
Comment (PatrykSzwer) [inline src/KitchenSink/wwwroot/KitchenSink/CheckboxPage.html]: Same as above.

File: src/KitchenSink/wwwroot/KitchenSink/CookiePage.html (hunk 21 above)
Comment (PatrykSzwer) [inline src/KitchenSink/wwwroot/KitchenSink/CookiePage.html]: Same as above.

File: src/KitchenSink/wwwroot/KitchenSink/ProgressBarPage.html (hunk 13 above)
Comment (PatrykSzwer) [inline src/KitchenSink/wwwroot/KitchenSink/ProgressBarPage.html]: I'm not sure about that one. Isn't that progress bar as percentage and fill is a complex think and should be placed/moved as one object? As entire progress bar, not its peaces.

File: src/KitchenSink/wwwroot/KitchenSink/BreadcrumbPage.html (hunk 15 above)
Comment (tomalec) [inline src/KitchenSink/wwwroot/KitchenSink/BreadcrumbPage.html]: > I'm pretty sure there was some kind of issue with putting slots on template elements, which is why that was never done in the People app

There shouldn't be, if there is one, please report.

> I've always done that div wrap in next files, in this one I didn't know that yet and forgot to change it later. I'll fix it.

Wrapping in a div makes slightly different behavior, are you sure ou want it here? I'd just add slot name to the `h5`

File: src/KitchenSink/wwwroot/KitchenSink/BreadcrumbPage.html (hunk 17 above)
Comment (tomalec) [inline src/KitchenSink/wwwroot/KitchenSink/BreadcrumbPage.html]: I have responded there.

File: src/KitchenSink/wwwroot/KitchenSink/ProgressBarPage.html (hunk 13 above)
Comment (tomalec) [inline src/KitchenSink/wwwroot/KitchenSink/ProgressBarPage.html]: I agree with @PatrykSzwer .
@Mackiovello I think we can reason about it like it's a just very simple and lightweight custom element (from pre-WC era). We shouldn't force app devs to use WC, so I'd say it's 100% fine.

Comment (Mackiovello) [inline src/KitchenSink/wwwroot/KitchenSink/ProgressBarPage.html]: Yea, that's a fair point

File: src/KitchenSink/wwwroot/KitchenSink/AutocompletePage.html (hunk 14 above)
Comment (tomalec) [inline src/KitchenSink/wwwroot/KitchenSink/AutocompletePage.html]: We would like to, but so far we didn't find a good alternative. Bootstrap is not playing nice with styles applied in Shadow DOM and is not as modular as we need.

**Comment by PatrykSzwer** [comment]
As a reference to discussion in #97 I removed unnecessary shado dom attachements. Right now I'm fighting with one test to adapt it to new changes.

File: src/KitchenSink/wwwroot/KitchenSink/DialogPage.html (hunk 9 above)
Comment (PatrykSzwer) [inline src/KitchenSink/wwwroot/KitchenSink/DialogPage.html]: Ok, I left <p> untouched. Lets look at this as a different issue.

File: src/KitchenSink/wwwroot/KitchenSink/FlashMessagePage.html (hunk 10 above)
Comment (PatrykSzwer) [inline src/KitchenSink/wwwroot/KitchenSink/FlashMessagePage.html]: As above.

File: src/KitchenSink/wwwroot/KitchenSink/AutocompletePage.html (hunk 14 above)
Comment (PatrykSzwer) [inline src/KitchenSink/wwwroot/KitchenSink/AutocompletePage.html]: So if I understand correctly, for now, I'm leaving that boostrap class untouched. Please confirm if everything else is acceptable for merge.

File: src/KitchenSink/wwwroot/KitchenSink/BreadcrumbPage.html [hunk 22]
```diff
@@ -37,18 +36,20 @@
...
-            <h5>Adding a new type in <strong>{{model.CurrentTreeItem.ParentName}}</strong></h5>
-        </template>
-
-        <template is="dom-if" if="{{!model.CurrentTreeItem.IsAdd}}" restamp>
-            <h5>Editing <strong>{{model.CurrentTreeItem.Name$}}</strong></h5>
-        </template>
-
-        <label class="control-label">Name</label>
+        <div slot="kitchensink/breadcrumb-adding-heading">
+            <template slot="kitchensink/breadcrumb-adding-heading" is="dom-if" if="{{model.CurrentTreeItem.IsAdd}}" restamp>
```
Comment (tomalec) [inline src/KitchenSink/wwwroot/KitchenSink/BreadcrumbPage.html]: You don't need slot for grandchildren, you add them to the elements that will (eventually) get stamped as children. Also, I would remove the wrapping div, not to render it if `dom-if` condition is not met.

So in general I'd remove `div`s from l.39 and l.44 then add `slot` name for `h5` in l.41 and l.46.

**Pull request merged**